include/simdjson/generic/numberparsing.h

```diff
@@ -42,6 +42,8 @@ simdjson_really_inline double to_double(uint64_t mantissa, uint64_t real_exponen
 // We assume that power is in the [smallest_power,
 // largest_power] interval: the caller is responsible for this check.
 simdjson_really_inline bool compute_float_64(int64_t power, uint64_t i, bool negative, double &d) {
+  assert(power>=simdjson::internal::smallest_power);
```

Comment:
I would be tempted to remove these asserts... this is not a function that is callable by our users. It is only ever called by us, and we should (it is our job) track down the calls and ensure that we guard them appropriately.

But it is just a mild preference. Note that some users compile in release mode with asserts (as part of generic policies). Thankfully, they won't do that with On Demand...  but they might with DOM. (I hope not.)

Reply:
I like asserts, but I agree the risk of having people having them active is not what we want. Instead, we have fuzz testing, unit tests and sanitizers. I will remove them.